api/v1alpha1/akodeploymentconfig_webhook.go

```diff
@@ -295,6 +295,9 @@ func (r *AKODeploymentConfig) validateAviServiceEngineGroup() *field.Error {
 // validateAviControlPlaneNetworks checks input Control Plane Network name existing or not, CIDR format valid or not
 func (r *AKODeploymentConfig) validateAviControlPlaneNetworks() field.ErrorList {
 	var allErrs field.ErrorList
+	if r.Spec.ControlPlaneNetwork.Name == "" || r.Spec.ControlPlaneNetwork.CIDR == "" {
```

Comment:
do we need to add a condition 'r.Spec.ControlPlaneNetwork == nil' in case of user does not configure the whole ControlPlaneNetwork object?

Reply:
Hi @honghaiw , that should be ok, even user doesn't configure this field, `r.Spec.ControlPlaneNetwork.Name` is a "" instead of a nil pointer cause we define it as a string instead of `*string`